src/FxCop/Desktop.Analyzers/Core/Security/DoNotUseInsecureCryptographicAlgorithmsAnalyzer.cs

```diff
@@ -127,38 +129,73 @@ public void AnalyzeNode(SyntaxNodeAnalysisContext context)
 
                 INamedTypeSymbol type = method.ContainingType;
                 DiagnosticDescriptor rule = null;
+                string[] messageArgs = new string[2];
+                string owningParentName = string.Empty;
+                SyntaxNode cur = node;
+
+                while(cur.Parent != null)
+                {
+                    var pNode = cur.Parent;
+                    ISymbol sym = SyntaxNodeHelper.GetSymbol(pNode, model);
+
+                    if(sym != null &&
+                        !string.IsNullOrEmpty(sym.Name)
+                        && (
+                            sym.Kind == SymbolKind.Method ||
+                            sym.Kind == SymbolKind.NamedType
+                           )
+                    )
+                    {
+                        owningParentName = sym.Name;
+                        break;
+                    }
+
+                    cur = pNode;
+                }
+
```

Comment:
I can only find a method GetEnclosingSymbol(), but it doesn't work for

await Task.Run(()=>new RC2CryptoServiceProvider());

The name of the symbol is empty here, which makes the message looks stupid.
I still need to have a while to walk up the tree. Any ideas?